routing/control_tower.go

```diff
@@ -294,4 +331,21 @@ func (p *controlTower) notifySubscribers(paymentHash lntypes.Hash,
 		case <-subscriber.quit:
```

Comment:
Since we are here, does anyone know why we don't remove the subscriber here?

Reply:
I don't know why exactly. I know the subscriber will be removed when the payment reaches a terminal state. And that the locks on subscribersMtx prevent complete parallel updates of payments. It may get pretty busy around that subscribersMtx, so I could imagine acquiring another lock there is unwanted. but that's a guess.